### DISCUSSION TITLE
Handlebars regex extension with slot item

### DISCUSSION THREAD

**Discussion by andikrueger** [microsoft-search/pnp-modern-search#2565] [Q&A]
**Is your feature request related to a problem? Please describe.**

I want to use a slot item and trim this one by using a regex. At the moment I fail at this point with the handle bar expression:

All of these do not work: 

{{regex "\d{4}" (slot item @root.slots.Jahr)}}

{{regex "\d{4}" {{slot item @root.slots.Jahr}} }}

The slotitem looks like this: 2022.000000000

**Comment by FranckyC** [comment]
Hi @andikrueger, does it work with the raw value (without the 'slot' expression)?

**Reply by andikrueger** [reply]
Yes. Raw value works:

{{regex "\d{4}" "2022.123123123" }} --> 2022
